tests/cephfs/cephfs_fscrypt/test_fscrypt_basic.py

```diff
@@ -0,0 +1,530 @@
+        for encrypt_dict in encrypt_path_list:
+            encrypt_params = encrypt_dict["encrypt_path_info"]["encrypt_params"]
+            protector_id = encrypt_params["protector_id"]
+            mnt_pt = mount_details[sv_name][mnt_type]["mountpoint"]
+            protector_params = {"id": protector_id}
+            protector_id = fscrypt_util.metadata_ops(
+                client1, "destroy", "protector", mnt_pt, **protector_params
+            )
+            policy_id = encrypt_params["policy_id"]
+            policy_params = {"id": policy_id}
+            policy_id = fscrypt_util.metadata_ops(
+                client1, "destroy", "policy", mnt_pt, **policy_params
+            )
+    if test_status:
+        return 1
+
+    return 0
+
+
+def fscrypt_non_empty_dir(fscrypt_test_params):
+    """
+    This is a testcase module to verify encrypt not allowed on non-empty directory path
+    Args:
+    Required:fscrypt_test_params in below format,
+    fscrypt_test_params = {
+            "fscrypt_util": fscrypt_util,
+            "clients": clients,
+            "mount_details": mount_details,
+        }
+    """
+    log.info("FScrypt test to verify encrypt not allowed on non-empty directory path")
+    fscrypt_util = fscrypt_test_params["fscrypt_util"]
+    clients = fscrypt_test_params["clients"]
+    mount_details = fscrypt_test_params["mount_details"]
+    client1 = clients[1]
+    log.info("Create test directory in one of subvolumes and add some data")
+    sv_name = random.choice(list(mount_details.keys()))
+    mnt_type = random.choice(list(mount_details[sv_name].keys()))
+    mountpoint = mount_details[sv_name][mnt_type]["mountpoint"]
+    encrypt_path = f"{mountpoint}/fscrypt_non_empty_dir"
+    cmd = f"mkdir {encrypt_path}"
+    client1.exec_command(
+        sudo=True,
+        cmd=cmd,
+        check_ec=False,
+    )
+    log.info("Add file and dir to encrypt dir path")
+    cmd = f"cd {encrypt_path}/;cp /var/log/messages .;"
+    cmd += "mkdir dir1;cp messages dir1/"
+    client1.exec_command(
+        sudo=True,
+        cmd=cmd,
+        check_ec=False,
+    )
+    log.info("fscrypt setup on mountpoint")
+    if fscrypt_util.setup(client1, mountpoint):
+        log.error(f"FScrypt setup on {mountpoint} failed for {sv_name}")
+        return 1
+
+    log.info("fscrypt encrypt on non-empty test directory")
+    try:
+        fscrypt_util.encrypt(client1, encrypt_path, mountpoint)
+        log.error(
+            "FScrypt suceeded on non_empty dir %s, it's not expected", encrypt_path
+        )
+        return 1
+    except BaseException as ex:
+        if "cannot be encrypted because it is non-empty" in str(ex):
+            log.info("FScrypt failed on non_empty encrypt path with expected error")
+            return 0
+        else:
+            log.error(
+                "FScrypt failed on non_empty encrypt path but with unexpected error"
+            )
+            return 1
```

Comment:
This is bit confusing to read as the default path is negative. Also, there is a possibility an `exception` could lead to negative path. Extending here would be difficult. Could we refactor to handling the exception?

It is not a good practise to handle generic exceptions.